package/MDAnalysis/lib/accessors.py

```diff
@@ -0,0 +1,145 @@
+# -*- Mode: python; tab-width: 4; indent-tabs-mode:nil; coding:utf-8 -*-
+# vim: tabstop=4 expandtab shiftwidth=4 softtabstop=4 fileencoding=utf-8
+#
+# MDAnalysis --- https://www.mdanalysis.org
+# Copyright (c) 2006-2017 The MDAnalysis Development Team and contributors
+# (see the file AUTHORS for the full list of names)
+#
+# Released under the GNU Public Licence, v2 or any higher version
+#
+# Please cite your use of MDAnalysis in published work:
+#
+# R. J. Gowers, M. Linke, J. Barnoud, T. J. E. Reddy, M. N. Melo, S. L. Seyler,
+# D. L. Dotson, J. Domanski, S. Buchoux, I. M. Kenney, and O. Beckstein.
+# MDAnalysis: A Python package for the rapid analysis of molecular dynamics
+# simulations. In S. Benthall and S. Rostrup editors, Proceedings of the 15th
+# Python in Science Conference, pages 102-109, Austin, TX, 2016. SciPy.
+# doi: 10.25080/majora-629e541a-00e
+#
+# N. Michaud-Agrawal, E. J. Denning, T. B. Woolf, and O. Beckstein.
+# MDAnalysis: A Toolkit for the Analysis of Molecular Dynamics Simulations.
+# J. Comput. Chem. 32 (2011), 2319--2327, doi:10.1002/jcc.21787
+#
+
+"""AtomGroup accessors --- :mod:`MDAnalysis.lib.accessors`
+=============================================================
+
+The aim of this module is simply to link a wrapper class to an object in order
+to avoid cluttering the namespace of the object.
+
+Example
+-------
+
+.. code-block:: python
+
+    >>> class SpeechWrapper:
+    ...     def __init__(self, person):
+    ...         self.person = person
+    ...     def __call__(self, *args):
+    ...         print(self.person.name, "says", *args)
+    ...     def whoami(self):
+    ...         print("I am %s" % self.person.name)
+    ...
+    >>> class Person:
+    ...     def __init__(self, name):
+    ...         self.name = name
+    ...     say = Accessor(SpeechWrapper)
+    ...
+    >>> bob = Person("Bob")
+    >>> bob.say("hello")
+    Bob says hello
+    >>> bob.say.whoami()
+    I am Bob
+
+"""
+
+from functools import partial, update_wrapper
+
+from .. import _CONVERTERS
+from ..core._get_readers import get_converter_for
```

Comment:
lib is not allowed to import from core or really anywhere outside lib – otherwise we end up with cyclical imports.

You can move accessors into a new `MDAnalysis.converters` package. At least then we have a good place to put all the machinery and docs.